extension/json/json_functions/json_transform.cpp

```diff
@@ -689,6 +689,55 @@ bool TransformToJSON(yyjson_val *vals[], yyjson_alc *alc, Vector &result, const
 	return true;
 }
 
+bool TransformValueIntoUnion(yyjson_val **vals, yyjson_alc *alc, Vector &result, const idx_t count,
+                             JSONTransformOptions &options) {
+	auto type = result.GetType();
+
+	auto fields = UnionType::CopyMemberTypes(type);
+	vector<string> names;
+	for (const auto &field : fields) {
+		names.push_back(field.first);
+	}
+
+	for (idx_t i = 0; i < count; i++) {
+		if (!yyjson_is_obj(vals[i])) {
+			if (options.error_missing_key) {
+				throw InvalidInputException("Expected an object representing a union");
+			} else {
+				result.SetValue(i, Value(nullptr));
+				continue;
+			}
+		}
+
+		yyjson_val *key, *val;
```

Comment:
I would recommend using `yyjson_obj_foreach` here, it makes the loop more readable. There's a few examples in this file.